### ISSUE TITLE
Improve zero_terms_query option

### ISSUE THREAD

**Issue by atsushi-matsui** [elastic/elasticsearch#110075]
### Overview
In search queries such as match_query, zero_terms_query only handles two types, "none" and "all", but I would like to be able to use the "null" defined [here](https://github.com/atsushi-matsui/elasticsearch/blob/13478b2bca6d8c3d523027ee7313e74ae1c65992/server/src/main/java/org/elasticsearch/index/query/ZeroTermsQueryOption.java#L33) as well.

#### Problem
when a query like "Quick the Brown Fox" is searched and spaces are treated as AND operators, constructing a Boolean query like the one below may result in zero hits.
The main cause of this is thought to be that the analyzer is setting the tokens to 0.
For example, an easy example of this would be when a stopwords filter is set to the analyzer.
```
{
  "query": {
    "bool": {
      "must": [
        {
          "match": {
            "title": "Quick"
          }
        },
        {
          "match": {
            "title": "the"
          }
        },
        {
          "match": {
            "title": "Brown"
          }
        },
        {
          "match": {
            "title": "Fox"
          }
        }
      ]
    }
  }
}
```

Also, if you set zero_terms_query to "all" in an attempt to avoid zero hits, you will end up hitting all results, even though you should actually get zero.
An example of a query is a boolean query that contains only stop words, like the one below.
```
{
  "query": {
    "bool": {
      "must": [
        {
          "match": {
            "title": "a",
            "zero_terms_query", "all"
          }
        },
        {
          "match": {
            "title": "the",
            "zero_terms_query", "all"
          }
        }
      ]
    }
  }
}
```
#### proposal
I would like to add a new option for zero_terms_query called "omit".
When set, it is expected that if the token becomes 0, the query will be ignored.
In other words, if a user searches for "Quick the Brown Fox", the word "the" will be ignored and a search query like "Quick AND Brown AND Fox" will be executed.
I think this behavior is already implemented in query_string and simple_query_string.

**Comment by atsushi-matsui** [comment]
After making [these changes](https://github.com/atsushi-matsui/elasticsearch/pull/1) and testing it locally, I got the results I expected.
If there are no problems, I will submit a PR.

**Comment by atsushi-matsui** [comment]
Close this issue because created [new issue](https://github.com/elastic/elasticsearch/issues/110080).
